internal/config/profile_test.go

```diff
@@ -0,0 +1,304 @@
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package config
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/spf13/afero"
+	"github.com/spf13/viper"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	atlasProfile   = "atlas"
+	newProfileName = "newProfileName"
+)
+
+var testConfigDir = "/test_dir"
+var fs afero.Fs
+var filename = fmt.Sprintf("%v/%v.toml", testConfigDir, ToolName)
+
+func TestMain(m *testing.M) {
+	fs = afero.NewMemMapFs()
+
+	viper.SetFs(fs)
+	SetConfigPath(testConfigDir)
+
+	code := m.Run()
+	os.Exit(code)
+}
+
+func createProfile(profileContents string) (*profile, error) {
+	err := afero.WriteFile(fs, filename, []byte(profileContents), 0600)
+	if err != nil {
+		return nil, err
+	}
+
+	return &profile{
+		name:      "default",
+		configDir: testConfigDir,
+		fs:        fs,
+	}, nil
+}
+
+func createProfileWithJustDefaultUser() (*profile, error) {
+	const contents = `
+[default]
+  base_url = "http://base_url.com/"
+  ops_manager_skip_verify = "false"
+  org_id = "5cac6a2179358edabd12b572"
+`
+	return createProfile(contents)
+}
+
+func createProfileWithOneNonDefaultUser() (*profile, error) {
+	const contents = `
+[atlas]
+  base_url = "http://base_url.com/"
+  ops_manager_skip_verify = "false"
+  org_id = "5cac6a2179358edabd12b572"
+`
+	return createProfile(contents)
+}
+
+func createProfileWithOneDefaultUserOneNonDefault() (*profile, error) {
+	const contents = `
+[default]
+  base_url = "http://base_url.com/"
+  ops_manager_skip_verify = "false"
+  org_id = "5cac6a2179358edabd12b572"
+  profile_id = "5cac6a2179358edabd12b572"
+
+[atlas]
+  base_url = "http://atlas.com/"
+  ops_manager_skip_verify = "false"
+  org_id = "5cac6a2179358edabd12b572"
+`
+	return createProfile(contents)
+}
+
+func createProfileWithFullDescription() (*profile, error) {
+	const contents = `
+[default]
+  base_url = "http://base_url.com/"
+  ops_manager_url = "http://om_url.com/"
+  ops_manager_ca_certificate = "/path/to/certificate"
+  ops_manager_skip_verify = "false"
+  public_api_key = "some_public_key"
+  private_api_key = "some_private_key"
+  org_id = "5cac6a2179358edabd12b572"
+  profile_id = "5cac6a2179358edabd12b572"
+  service = "cloud"
+`
+	return createProfile(contents)
+}
+
+func TestProfileList(t *testing.T) {
+	profile, err := createProfileWithOneDefaultUserOneNonDefault()
+	if err != nil {
+		t.Error(err)
+	}
+
+	if err := profile.Load(false); err != nil {
+		t.Error(err)
+	}
+
+	availableProfiles := profile.List()
+	assert.Equal(t, 2, len(availableProfiles), "expected to find 2 profiles")
+}
+
+func TestProfileDescribeFullProfile(t *testing.T) {
```

Comment:
[nit-style] your test method should match the method name, since you're testing the struct profile, and its method 
`GetConfigDescription` this test should be 
```suggestion
func TestProfile_GetConfigDescription(t *testing.T) {
```
if you have multiple scenarios you can add them like `TestProfile_GetConfigDescription_FullProfile` tho I prefer to use internal functions/closures  like `t.Run("full profile", func(t *testing.T) {` the second let's you do some basic scenarios as code kind of thing, you can check [this test](https://github.com/mongodb/mongocli/blob/f480e0f9ca05bdb5e9177029f38bc3dcc9a97657/internal/convert/cluster_config_test.go#L25) for an example of what I mean (see how I define a name, initial state, changes, expected map and then apply the `t.Run` function to each scenario